### PR TITLE
Link db logs

### PR DESCRIPTION
**Pull request by AxelHumbert**

### CODE REVIEW

**Review by Idhao — APPROVED** [review]

File: src/main/java/fr/uge/chargepointconfiguration/controller/BusinessLogController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   * BusinessLogController's constructor.
+   *
+   * @param businessLogRepository A BusinessLogRepository.
+   */
+  @Autowired
+  public BusinessLogController(BusinessLogRepository businessLogRepository) {
+    this.businessLogRepository = businessLogRepository;
+  }
+
+  /**
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/controller/BusinessLogController.java:34]: replace by a read in db, we won't create logs from the front
Also you miss a @GetMapping or equivalent

File: src/main/java/fr/uge/chargepointconfiguration/controller/TechnicalLogController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   * @param technicalLogRepository a TechnicalLogRepository.
+   */
+  @Autowired
+  public TechnicalLogController(TechnicalLogRepository technicalLogRepository) {
+    this.technicalLogRepository = technicalLogRepository;
+  }
+
+  /**
+   * Save the technical log in the database.
+   *
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/controller/TechnicalLogController.java:36]: same here

File: src/main/java/fr/uge/chargepointconfiguration/entities/BusinessLog.java [hunk 3]
```diff
@@ -25,9 +26,10 @@ public class BusinessLog {
 
   @Column(name = "date", nullable = false,
           columnDefinition = "datetime default current_timestamp")
+  @CreationTimestamp
   private Timestamp date;
 
-  @Column(name = "user", nullable = false, length = 45)
+  @Column(name = "\"user\"", nullable = false, length = 45)
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/entities/BusinessLog.java:36]: comment to explain the quotes inside "user"

File: src/main/java/fr/uge/chargepointconfiguration/repository/BusinessLogRepository.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import fr.uge.chargepointconfiguration.entities.BusinessLog;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Repository for the business log.
+ */
+@Repository
+public interface BusinessLogRepository extends CrudRepository<BusinessLog, Integer> {
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/repository/BusinessLogRepository.java:12]: don't forget to have some methods returning a List

File: src/main/java/fr/uge/chargepointconfiguration/repository/TechnicalLogRepository.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import fr.uge.chargepointconfiguration.entities.TechnicalLog;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Repository for technical log.
+ */
+@Repository
+public interface TechnicalLogRepository extends CrudRepository<TechnicalLog, Integer> {
+
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/repository/TechnicalLogRepository.java:13]: same here

**Review by Adrien-COSTANDI — CHANGES_REQUESTED** [review]
few changes

File: src/main/java/fr/uge/chargepointconfiguration/entities/BusinessLog.java (hunk 3 above)
Comment (Idhao) [reply src/main/java/fr/uge/chargepointconfiguration/entities/BusinessLog.java:36]: Then do the same for User JPA class

File: src/main/java/fr/uge/chargepointconfiguration/controller/BusinessLogController.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+   * Returns a list of business logs according to the given charge point.
+   *
+   * @param chargepoint the id of the charge point.
+   * @return a list of business logs by charge point.
+   */
+  @Operation(summary = "Get a list of logs by its charge point")
+  @ApiResponse(responseCode = "200",
+          description = "Found the list of business logs",
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/controller/BusinessLogController.java:42]: Retrieve the list*

**Pull request merged**